Testing master fails: `Lingering task` 3x

Error in Testing and Coverage(Testing):

```
2026-05-03T08:02:08.9083492Z tests/tests_new/test_services.py .............E.E....................... [ 59%]
2026-05-03T08:02:10.2642556Z .................................................E                       [ 69%]
(...)
2026-05-03T08:02:49.7206446Z ==================================== ERRORS ====================================
2026-05-03T08:02:49.7206954Z ________________ ERROR at teardown of test_bind_device_success _________________
2026-05-03T08:02:49.7207286Z 
2026-05-03T08:02:49.7207474Z expected_lingering_tasks = False, expected_lingering_timers = False
2026-05-03T08:02:49.7207751Z 
2026-05-03T08:02:49.7207857Z     @pytest_asyncio.fixture(autouse=True)
2026-05-03T08:02:49.7208119Z     def verify_cleanup(
2026-05-03T08:02:49.7208338Z         expected_lingering_tasks: bool,
2026-05-03T08:02:49.7208602Z         expected_lingering_timers: bool,
2026-05-03T08:02:49.7208880Z     ) -> Generator[None]:
2026-05-03T08:02:49.7209165Z         """Verify that the test has cleaned up resources correctly.
2026-05-03T08:02:49.7209857Z     
2026-05-03T08:02:49.7210079Z         This fixture requires the event loop to be stopped.
2026-05-03T08:02:49.7210395Z         It therefore cannot be an async fixture.
2026-05-03T08:02:49.7210634Z     
2026-05-03T08:02:49.7210909Z         Use @pytest_asyncio.fixture to make sure the correct event loop is set
2026-05-03T08:02:49.7211276Z         regardless before calling the fixture.
2026-05-03T08:02:49.7211905Z         """
2026-05-03T08:02:49.7212106Z         event_loop = asyncio.get_event_loop()
2026-05-03T08:02:49.7212413Z         threads_before = frozenset(threading.enumerate())
2026-05-03T08:02:49.7212724Z         tasks_before = asyncio.all_tasks(event_loop)
2026-05-03T08:02:49.7212980Z         yield
2026-05-03T08:02:49.7213138Z     
2026-05-03T08:02:49.7213391Z         event_loop.run_until_complete(event_loop.shutdown_default_executor())
2026-05-03T08:02:49.7213723Z     
2026-05-03T08:02:49.7213880Z         if len(INSTANCES) >= 2:
2026-05-03T08:02:49.7214107Z             count = len(INSTANCES)
2026-05-03T08:02:49.7214327Z             for inst in INSTANCES:
2026-05-03T08:02:49.7214542Z                 inst.stop()
2026-05-03T08:02:49.7214868Z             pytest.exit(f"Detected non stopped instances ({count}), aborting test run")
2026-05-03T08:02:49.7215221Z     
2026-05-03T08:02:49.7215454Z         # Warn and clean-up lingering tasks and timers
2026-05-03T08:02:49.7215738Z         # before moving on to the next test.
2026-05-03T08:02:49.7216051Z         tasks = asyncio.all_tasks(event_loop) - tasks_before
2026-05-03T08:02:49.7216330Z         for task in tasks:
2026-05-03T08:02:49.7216549Z             if expected_lingering_tasks:
2026-05-03T08:02:49.7216848Z                 _LOGGER.warning("Lingering task after test %r", task)
2026-05-03T08:02:49.7217139Z             else:
2026-05-03T08:02:49.7217371Z                 pytest.fail(f"Lingering task after test {task!r}")
2026-05-03T08:02:49.7217659Z             task.cancel()
2026-05-03T08:02:49.7217850Z         if tasks:
2026-05-03T08:02:49.7218076Z             event_loop.run_until_complete(asyncio.wait(tasks))
2026-05-03T08:02:49.7218348Z     
2026-05-03T08:02:49.7218554Z         for handle in get_scheduled_timer_handles(event_loop):
2026-05-03T08:02:49.7218841Z             if not handle.cancelled():
2026-05-03T08:02:49.7219080Z                 with long_repr_strings():
2026-05-03T08:02:49.7219550Z                     if expected_lingering_timers:
2026-05-03T08:02:49.7219887Z                         _LOGGER.warning("Lingering timer after test %r", handle)
2026-05-03T08:02:49.7220507Z                     elif handle._args and isinstance(job := handle._args[-1], HassJob):
2026-05-03T08:02:49.7220863Z                         if job.cancel_on_shutdown:
2026-05-03T08:02:49.7221109Z                             continue
2026-05-03T08:02:49.7221398Z >                       pytest.fail(f"Lingering timer after job {job!r}")
2026-05-03T08:02:49.7222306Z E                       Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad74ac30>>>
2026-05-03T08:02:49.7223093Z 
2026-05-03T08:02:49.7223506Z /opt/hostedtoolcache/Python/3.14.4/x64/lib/python3.14/site-packages/pytest_homeassistant_custom_component/plugins.py:436: Failed
2026-05-03T08:02:49.7224132Z ----------------------------- Captured stderr call -----------------------------
2026-05-03T08:02:49.7224635Z WARNING:custom_components.ramses_cc.services:Starting binding process for device 01:123456
2026-05-03T08:02:49.7225235Z WARNING:custom_components.ramses_cc.services:Success! Binding process completed for device 01:123456
2026-05-03T08:02:49.7225744Z ------------------------------ Captured log call -------------------------------
2026-05-03T08:02:49.7226264Z WARNING  custom_components.ramses_cc.services:services.py:81 Starting binding process for device 01:123456
2026-05-03T08:02:49.7226935Z WARNING  custom_components.ramses_cc.services:services.py:94 Success! Binding process completed for device 01:123456
2026-05-03T08:02:49.7227493Z _______________ ERROR at teardown of test_send_packet_hgi_alias ________________
2026-05-03T08:02:49.7227758Z 
2026-05-03T08:02:49.7227939Z expected_lingering_tasks = False, expected_lingering_timers = False
2026-05-03T08:02:49.7228189Z 
2026-05-03T08:02:49.7228290Z     @pytest_asyncio.fixture(autouse=True)
2026-05-03T08:02:49.7228533Z     def verify_cleanup(
2026-05-03T08:02:49.7228892Z         expected_lingering_tasks: bool,
2026-05-03T08:02:49.7229143Z         expected_lingering_timers: bool,
2026-05-03T08:02:49.7229692Z     ) -> Generator[None]:
2026-05-03T08:02:49.7229971Z         """Verify that the test has cleaned up resources correctly.
2026-05-03T08:02:49.7230252Z     
2026-05-03T08:02:49.7230464Z         This fixture requires the event loop to be stopped.
2026-05-03T08:02:49.7230765Z         It therefore cannot be an async fixture.
2026-05-03T08:02:49.7231008Z     
2026-05-03T08:02:49.7231263Z         Use @pytest_asyncio.fixture to make sure the correct event loop is set
2026-05-03T08:02:49.7231631Z         regardless before calling the fixture.
2026-05-03T08:02:49.7231875Z         """
2026-05-03T08:02:49.7232062Z         event_loop = asyncio.get_event_loop()
2026-05-03T08:02:49.7232357Z         threads_before = frozenset(threading.enumerate())
2026-05-03T08:02:49.7232666Z         tasks_before = asyncio.all_tasks(event_loop)
2026-05-03T08:02:49.7232930Z         yield
2026-05-03T08:02:49.7233087Z     
2026-05-03T08:02:49.7233348Z         event_loop.run_until_complete(event_loop.shutdown_default_executor())
2026-05-03T08:02:49.7233668Z     
2026-05-03T08:02:49.7233835Z         if len(INSTANCES) >= 2:
2026-05-03T08:02:49.7234060Z             count = len(INSTANCES)
2026-05-03T08:02:49.7234280Z             for inst in INSTANCES:
2026-05-03T08:02:49.7234496Z                 inst.stop()
2026-05-03T08:02:49.7234821Z             pytest.exit(f"Detected non stopped instances ({count}), aborting test run")
2026-05-03T08:02:49.7235171Z     
2026-05-03T08:02:49.7235364Z         # Warn and clean-up lingering tasks and timers
2026-05-03T08:02:49.7235646Z         # before moving on to the next test.
2026-05-03T08:02:49.7235934Z         tasks = asyncio.all_tasks(event_loop) - tasks_before
2026-05-03T08:02:49.7236217Z         for task in tasks:
2026-05-03T08:02:49.7236433Z             if expected_lingering_tasks:
2026-05-03T08:02:49.7236733Z                 _LOGGER.warning("Lingering task after test %r", task)
2026-05-03T08:02:49.7237188Z             else:
2026-05-03T08:02:49.7237416Z                 pytest.fail(f"Lingering task after test {task!r}")
2026-05-03T08:02:49.7237707Z             task.cancel()
2026-05-03T08:02:49.7237890Z         if tasks:
2026-05-03T08:02:49.7238122Z             event_loop.run_until_complete(asyncio.wait(tasks))
2026-05-03T08:02:49.7238386Z     
2026-05-03T08:02:49.7238596Z         for handle in get_scheduled_timer_handles(event_loop):
2026-05-03T08:02:49.7238888Z             if not handle.cancelled():
2026-05-03T08:02:49.7239125Z                 with long_repr_strings():
2026-05-03T08:02:49.7239501Z                     if expected_lingering_timers:
2026-05-03T08:02:49.7239824Z                         _LOGGER.warning("Lingering timer after test %r", handle)
2026-05-03T08:02:49.7240231Z                     elif handle._args and isinstance(job := handle._args[-1], HassJob):
2026-05-03T08:02:49.7240577Z                         if job.cancel_on_shutdown:
2026-05-03T08:02:49.7240840Z                             continue
2026-05-03T08:02:49.7241110Z >                       pytest.fail(f"Lingering timer after job {job!r}")
2026-05-03T08:02:49.7242018Z E                       Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad732150>>>
2026-05-03T08:02:49.7242776Z 
2026-05-03T08:02:49.7243150Z /opt/hostedtoolcache/Python/3.14.4/x64/lib/python3.14/site-packages/pytest_homeassistant_custom_component/plugins.py:436: Failed
2026-05-03T08:02:49.7243773Z ________ ERROR at teardown of test_async_bind_device_routes_to_registry ________
2026-05-03T08:02:49.7244045Z 
2026-05-03T08:02:49.7244214Z expected_lingering_tasks = False, expected_lingering_timers = False
2026-05-03T08:02:49.7244462Z 
2026-05-03T08:02:49.7244561Z     @pytest_asyncio.fixture(autouse=True)
2026-05-03T08:02:49.7244797Z     def verify_cleanup(
2026-05-03T08:02:49.7245144Z         expected_lingering_tasks: bool,
2026-05-03T08:02:49.7245407Z         expected_lingering_timers: bool,
2026-05-03T08:02:49.7245653Z     ) -> Generator[None]:
2026-05-03T08:02:49.7245908Z         """Verify that the test has cleaned up resources correctly.
2026-05-03T08:02:49.7246190Z     
2026-05-03T08:02:49.7246398Z         This fixture requires the event loop to be stopped.
2026-05-03T08:02:49.7246697Z         It therefore cannot be an async fixture.
2026-05-03T08:02:49.7246933Z     
2026-05-03T08:02:49.7247183Z         Use @pytest_asyncio.fixture to make sure the correct event loop is set
2026-05-03T08:02:49.7247541Z         regardless before calling the fixture.
2026-05-03T08:02:49.7247766Z         """
2026-05-03T08:02:49.7247956Z         event_loop = asyncio.get_event_loop()
2026-05-03T08:02:49.7248235Z         threads_before = frozenset(threading.enumerate())
2026-05-03T08:02:49.7248547Z         tasks_before = asyncio.all_tasks(event_loop)
2026-05-03T08:02:49.7248802Z         yield
2026-05-03T08:02:49.7248966Z     
2026-05-03T08:02:49.7249218Z         event_loop.run_until_complete(event_loop.shutdown_default_executor())
2026-05-03T08:02:49.7249843Z     
2026-05-03T08:02:49.7250006Z         if len(INSTANCES) >= 2:
2026-05-03T08:02:49.7250221Z             count = len(INSTANCES)
2026-05-03T08:02:49.7250441Z             for inst in INSTANCES:
2026-05-03T08:02:49.7250649Z                 inst.stop()
2026-05-03T08:02:49.7250968Z             pytest.exit(f"Detected non stopped instances ({count}), aborting test run")
2026-05-03T08:02:49.7251310Z     
2026-05-03T08:02:49.7251522Z         # Warn and clean-up lingering tasks and timers
2026-05-03T08:02:49.7251800Z         # before moving on to the next test.
2026-05-03T08:02:49.7252084Z         tasks = asyncio.all_tasks(event_loop) - tasks_before
2026-05-03T08:02:49.7252363Z         for task in tasks:
2026-05-03T08:02:49.7252587Z             if expected_lingering_tasks:
2026-05-03T08:02:49.7252888Z                 _LOGGER.warning("Lingering task after test %r", task)
2026-05-03T08:02:49.7253356Z             else:
2026-05-03T08:02:49.7253592Z                 pytest.fail(f"Lingering task after test {task!r}")
2026-05-03T08:02:49.7253885Z             task.cancel()
2026-05-03T08:02:49.7254071Z         if tasks:
2026-05-03T08:02:49.7254304Z             event_loop.run_until_complete(asyncio.wait(tasks))
2026-05-03T08:02:49.7254571Z     
2026-05-03T08:02:49.7254787Z         for handle in get_scheduled_timer_handles(event_loop):
2026-05-03T08:02:49.7255074Z             if not handle.cancelled():
2026-05-03T08:02:49.7255317Z                 with long_repr_strings():
2026-05-03T08:02:49.7255565Z                     if expected_lingering_timers:
2026-05-03T08:02:49.7255892Z                         _LOGGER.warning("Lingering timer after test %r", handle)
2026-05-03T08:02:49.7256307Z                     elif handle._args and isinstance(job := handle._args[-1], HassJob):
2026-05-03T08:02:49.7256659Z                         if job.cancel_on_shutdown:
2026-05-03T08:02:49.7256916Z                             continue
2026-05-03T08:02:49.7257188Z >                       pytest.fail(f"Lingering timer after job {job!r}")
2026-05-03T08:02:49.7258091Z E                       Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad7b28d0>>>
2026-05-03T08:02:49.7258842Z 
2026-05-03T08:02:49.7259216Z /opt/hostedtoolcache/Python/3.14.4/x64/lib/python3.14/site-packages/pytest_homeassistant_custom_component/plugins.py:436: Failed
2026-05-03T08:02:49.7259926Z ----------------------------- Captured stderr call -----------------------------
2026-05-03T08:02:49.7260419Z WARNING:custom_components.ramses_cc.services:Starting binding process for device 01:123456
2026-05-03T08:02:49.7261006Z WARNING:custom_components.ramses_cc.services:Success! Binding process completed for device 01:123456
2026-05-03T08:02:49.7261512Z ------------------------------ Captured log call -------------------------------
2026-05-03T08:02:49.7262297Z WARNING  custom_components.ramses_cc.services:services.py:81 Starting binding process for device 01:123456
2026-05-03T08:02:49.7262969Z WARNING  custom_components.ramses_cc.services:services.py:94 Success! Binding process completed for device 01:123456
2026-05-03T08:02:49.7263522Z --------------------------- snapshot report summary ----------------------------
2026-05-03T08:02:49.7263842Z 3 snapshots passed.
2026-05-03T08:02:49.7264086Z =========================== short test summary info ============================
2026-05-03T08:02:49.7265282Z ERROR tests/tests_new/test_services.py::test_bind_device_success - Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad74ac30>>>
2026-05-03T08:02:49.7267018Z ERROR tests/tests_new/test_services.py::test_send_packet_hgi_alias - Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad732150>>>
2026-05-03T08:02:49.7268804Z ERROR tests/tests_new/test_services.py::test_async_bind_device_routes_to_registry - Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad7b28d0>>>
2026-05-03T08:02:49.7270149Z ============= 501 passed, 10 skipped, 3 errors in 65.25s (0:01:05) =============
2026-05-03T08:02:49.8419041Z ##[error]Process completed with exit code 1.
2026-05-03T08:02:08.9083492Z tests/tests_new/test_services.py .............E.E....................... [ 59%]
2026-05-03T08:02:10.2642556Z .................................................E                       [ 69%]
(...)
2026-05-03T08:02:49.7206446Z ==================================== ERRORS ====================================
2026-05-03T08:02:49.7206954Z ________________ ERROR at teardown of test_bind_device_success _________________
2026-05-03T08:02:49.7207286Z 
2026-05-03T08:02:49.7207474Z expected_lingering_tasks = False, expected_lingering_timers = False
2026-05-03T08:02:49.7207751Z 
2026-05-03T08:02:49.7207857Z     @pytest_asyncio.fixture(autouse=True)
2026-05-03T08:02:49.7208119Z     def verify_cleanup(
2026-05-03T08:02:49.7208338Z         expected_lingering_tasks: bool,
2026-05-03T08:02:49.7208602Z         expected_lingering_timers: bool,
2026-05-03T08:02:49.7208880Z     ) -> Generator[None]:
2026-05-03T08:02:49.7209165Z         """Verify that the test has cleaned up resources correctly.
2026-05-03T08:02:49.7209857Z     
2026-05-03T08:02:49.7210079Z         This fixture requires the event loop to be stopped.
2026-05-03T08:02:49.7210395Z         It therefore cannot be an async fixture.
2026-05-03T08:02:49.7210634Z     
2026-05-03T08:02:49.7210909Z         Use @pytest_asyncio.fixture to make sure the correct event loop is set
2026-05-03T08:02:49.7211276Z         regardless before calling the fixture.
2026-05-03T08:02:49.7211905Z         """
2026-05-03T08:02:49.7212106Z         event_loop = asyncio.get_event_loop()
2026-05-03T08:02:49.7212413Z         threads_before = frozenset(threading.enumerate())
2026-05-03T08:02:49.7212724Z         tasks_before = asyncio.all_tasks(event_loop)
2026-05-03T08:02:49.7212980Z         yield
2026-05-03T08:02:49.7213138Z     
2026-05-03T08:02:49.7213391Z         event_loop.run_until_complete(event_loop.shutdown_default_executor())
2026-05-03T08:02:49.7213723Z     
2026-05-03T08:02:49.7213880Z         if len(INSTANCES) >= 2:
2026-05-03T08:02:49.7214107Z             count = len(INSTANCES)
2026-05-03T08:02:49.7214327Z             for inst in INSTANCES:
2026-05-03T08:02:49.7214542Z                 inst.stop()
2026-05-03T08:02:49.7214868Z             pytest.exit(f"Detected non stopped instances ({count}), aborting test run")
2026-05-03T08:02:49.7215221Z     
2026-05-03T08:02:49.7215454Z         # Warn and clean-up lingering tasks and timers
2026-05-03T08:02:49.7215738Z         # before moving on to the next test.
2026-05-03T08:02:49.7216051Z         tasks = asyncio.all_tasks(event_loop) - tasks_before
2026-05-03T08:02:49.7216330Z         for task in tasks:
2026-05-03T08:02:49.7216549Z             if expected_lingering_tasks:
2026-05-03T08:02:49.7216848Z                 _LOGGER.warning("Lingering task after test %r", task)
2026-05-03T08:02:49.7217139Z             else:
2026-05-03T08:02:49.7217371Z                 pytest.fail(f"Lingering task after test {task!r}")
2026-05-03T08:02:49.7217659Z             task.cancel()
2026-05-03T08:02:49.7217850Z         if tasks:
2026-05-03T08:02:49.7218076Z             event_loop.run_until_complete(asyncio.wait(tasks))
2026-05-03T08:02:49.7218348Z     
2026-05-03T08:02:49.7218554Z         for handle in get_scheduled_timer_handles(event_loop):
2026-05-03T08:02:49.7218841Z             if not handle.cancelled():
2026-05-03T08:02:49.7219080Z                 with long_repr_strings():
2026-05-03T08:02:49.7219550Z                     if expected_lingering_timers:
2026-05-03T08:02:49.7219887Z                         _LOGGER.warning("Lingering timer after test %r", handle)
2026-05-03T08:02:49.7220507Z                     elif handle._args and isinstance(job := handle._args[-1], HassJob):
2026-05-03T08:02:49.7220863Z                         if job.cancel_on_shutdown:
2026-05-03T08:02:49.7221109Z                             continue
2026-05-03T08:02:49.7221398Z >                       pytest.fail(f"Lingering timer after job {job!r}")
2026-05-03T08:02:49.7222306Z E                       Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad74ac30>>>
2026-05-03T08:02:49.7223093Z 
2026-05-03T08:02:49.7223506Z /opt/hostedtoolcache/Python/3.14.4/x64/lib/python3.14/site-packages/pytest_homeassistant_custom_component/plugins.py:436: Failed
2026-05-03T08:02:49.7224132Z ----------------------------- Captured stderr call -----------------------------
2026-05-03T08:02:49.7224635Z WARNING:custom_components.ramses_cc.services:Starting binding process for device 01:123456
2026-05-03T08:02:49.7225235Z WARNING:custom_components.ramses_cc.services:Success! Binding process completed for device 01:123456
2026-05-03T08:02:49.7225744Z ------------------------------ Captured log call -------------------------------
2026-05-03T08:02:49.7226264Z WARNING  custom_components.ramses_cc.services:services.py:81 Starting binding process for device 01:123456
2026-05-03T08:02:49.7226935Z WARNING  custom_components.ramses_cc.services:services.py:94 Success! Binding process completed for device 01:123456
2026-05-03T08:02:49.7227493Z _______________ ERROR at teardown of test_send_packet_hgi_alias ________________
2026-05-03T08:02:49.7227758Z 
2026-05-03T08:02:49.7227939Z expected_lingering_tasks = False, expected_lingering_timers = False
2026-05-03T08:02:49.7228189Z 
2026-05-03T08:02:49.7228290Z     @pytest_asyncio.fixture(autouse=True)
2026-05-03T08:02:49.7228533Z     def verify_cleanup(
2026-05-03T08:02:49.7228892Z         expected_lingering_tasks: bool,
2026-05-03T08:02:49.7229143Z         expected_lingering_timers: bool,
2026-05-03T08:02:49.7229692Z     ) -> Generator[None]:
2026-05-03T08:02:49.7229971Z         """Verify that the test has cleaned up resources correctly.
2026-05-03T08:02:49.7230252Z     
2026-05-03T08:02:49.7230464Z         This fixture requires the event loop to be stopped.
2026-05-03T08:02:49.7230765Z         It therefore cannot be an async fixture.
2026-05-03T08:02:49.7231008Z     
2026-05-03T08:02:49.7231263Z         Use @pytest_asyncio.fixture to make sure the correct event loop is set
2026-05-03T08:02:49.7231631Z         regardless before calling the fixture.
2026-05-03T08:02:49.7231875Z         """
2026-05-03T08:02:49.7232062Z         event_loop = asyncio.get_event_loop()
2026-05-03T08:02:49.7232357Z         threads_before = frozenset(threading.enumerate())
2026-05-03T08:02:49.7232666Z         tasks_before = asyncio.all_tasks(event_loop)
2026-05-03T08:02:49.7232930Z         yield
2026-05-03T08:02:49.7233087Z     
2026-05-03T08:02:49.7233348Z         event_loop.run_until_complete(event_loop.shutdown_default_executor())
2026-05-03T08:02:49.7233668Z     
2026-05-03T08:02:49.7233835Z         if len(INSTANCES) >= 2:
2026-05-03T08:02:49.7234060Z             count = len(INSTANCES)
2026-05-03T08:02:49.7234280Z             for inst in INSTANCES:
2026-05-03T08:02:49.7234496Z                 inst.stop()
2026-05-03T08:02:49.7234821Z             pytest.exit(f"Detected non stopped instances ({count}), aborting test run")
2026-05-03T08:02:49.7235171Z     
2026-05-03T08:02:49.7235364Z         # Warn and clean-up lingering tasks and timers
2026-05-03T08:02:49.7235646Z         # before moving on to the next test.
2026-05-03T08:02:49.7235934Z         tasks = asyncio.all_tasks(event_loop) - tasks_before
2026-05-03T08:02:49.7236217Z         for task in tasks:
2026-05-03T08:02:49.7236433Z             if expected_lingering_tasks:
2026-05-03T08:02:49.7236733Z                 _LOGGER.warning("Lingering task after test %r", task)
2026-05-03T08:02:49.7237188Z             else:
2026-05-03T08:02:49.7237416Z                 pytest.fail(f"Lingering task after test {task!r}")
2026-05-03T08:02:49.7237707Z             task.cancel()
2026-05-03T08:02:49.7237890Z         if tasks:
2026-05-03T08:02:49.7238122Z             event_loop.run_until_complete(asyncio.wait(tasks))
2026-05-03T08:02:49.7238386Z     
2026-05-03T08:02:49.7238596Z         for handle in get_scheduled_timer_handles(event_loop):
2026-05-03T08:02:49.7238888Z             if not handle.cancelled():
2026-05-03T08:02:49.7239125Z                 with long_repr_strings():
2026-05-03T08:02:49.7239501Z                     if expected_lingering_timers:
2026-05-03T08:02:49.7239824Z                         _LOGGER.warning("Lingering timer after test %r", handle)
2026-05-03T08:02:49.7240231Z                     elif handle._args and isinstance(job := handle._args[-1], HassJob):
2026-05-03T08:02:49.7240577Z                         if job.cancel_on_shutdown:
2026-05-03T08:02:49.7240840Z                             continue
2026-05-03T08:02:49.7241110Z >                       pytest.fail(f"Lingering timer after job {job!r}")
2026-05-03T08:02:49.7242018Z E                       Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad732150>>>
2026-05-03T08:02:49.7242776Z 
2026-05-03T08:02:49.7243150Z /opt/hostedtoolcache/Python/3.14.4/x64/lib/python3.14/site-packages/pytest_homeassistant_custom_component/plugins.py:436: Failed
2026-05-03T08:02:49.7243773Z ________ ERROR at teardown of test_async_bind_device_routes_to_registry ________
2026-05-03T08:02:49.7244045Z 
2026-05-03T08:02:49.7244214Z expected_lingering_tasks = False, expected_lingering_timers = False
2026-05-03T08:02:49.7244462Z 
2026-05-03T08:02:49.7244561Z     @pytest_asyncio.fixture(autouse=True)
2026-05-03T08:02:49.7244797Z     def verify_cleanup(
2026-05-03T08:02:49.7245144Z         expected_lingering_tasks: bool,
2026-05-03T08:02:49.7245407Z         expected_lingering_timers: bool,
2026-05-03T08:02:49.7245653Z     ) -> Generator[None]:
2026-05-03T08:02:49.7245908Z         """Verify that the test has cleaned up resources correctly.
2026-05-03T08:02:49.7246190Z     
2026-05-03T08:02:49.7246398Z         This fixture requires the event loop to be stopped.
2026-05-03T08:02:49.7246697Z         It therefore cannot be an async fixture.
2026-05-03T08:02:49.7246933Z     
2026-05-03T08:02:49.7247183Z         Use @pytest_asyncio.fixture to make sure the correct event loop is set
2026-05-03T08:02:49.7247541Z         regardless before calling the fixture.
2026-05-03T08:02:49.7247766Z         """
2026-05-03T08:02:49.7247956Z         event_loop = asyncio.get_event_loop()
2026-05-03T08:02:49.7248235Z         threads_before = frozenset(threading.enumerate())
2026-05-03T08:02:49.7248547Z         tasks_before = asyncio.all_tasks(event_loop)
2026-05-03T08:02:49.7248802Z         yield
2026-05-03T08:02:49.7248966Z     
2026-05-03T08:02:49.7249218Z         event_loop.run_until_complete(event_loop.shutdown_default_executor())
2026-05-03T08:02:49.7249843Z     
2026-05-03T08:02:49.7250006Z         if len(INSTANCES) >= 2:
2026-05-03T08:02:49.7250221Z             count = len(INSTANCES)
2026-05-03T08:02:49.7250441Z             for inst in INSTANCES:
2026-05-03T08:02:49.7250649Z                 inst.stop()
2026-05-03T08:02:49.7250968Z             pytest.exit(f"Detected non stopped instances ({count}), aborting test run")
2026-05-03T08:02:49.7251310Z     
2026-05-03T08:02:49.7251522Z         # Warn and clean-up lingering tasks and timers
2026-05-03T08:02:49.7251800Z         # before moving on to the next test.
2026-05-03T08:02:49.7252084Z         tasks = asyncio.all_tasks(event_loop) - tasks_before
2026-05-03T08:02:49.7252363Z         for task in tasks:
2026-05-03T08:02:49.7252587Z             if expected_lingering_tasks:
2026-05-03T08:02:49.7252888Z                 _LOGGER.warning("Lingering task after test %r", task)
2026-05-03T08:02:49.7253356Z             else:
2026-05-03T08:02:49.7253592Z                 pytest.fail(f"Lingering task after test {task!r}")
2026-05-03T08:02:49.7253885Z             task.cancel()
2026-05-03T08:02:49.7254071Z         if tasks:
2026-05-03T08:02:49.7254304Z             event_loop.run_until_complete(asyncio.wait(tasks))
2026-05-03T08:02:49.7254571Z     
2026-05-03T08:02:49.7254787Z         for handle in get_scheduled_timer_handles(event_loop):
2026-05-03T08:02:49.7255074Z             if not handle.cancelled():
2026-05-03T08:02:49.7255317Z                 with long_repr_strings():
2026-05-03T08:02:49.7255565Z                     if expected_lingering_timers:
2026-05-03T08:02:49.7255892Z                         _LOGGER.warning("Lingering timer after test %r", handle)
2026-05-03T08:02:49.7256307Z                     elif handle._args and isinstance(job := handle._args[-1], HassJob):
2026-05-03T08:02:49.7256659Z                         if job.cancel_on_shutdown:
2026-05-03T08:02:49.7256916Z                             continue
2026-05-03T08:02:49.7257188Z >                       pytest.fail(f"Lingering timer after job {job!r}")
2026-05-03T08:02:49.7258091Z E                       Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad7b28d0>>>
2026-05-03T08:02:49.7258842Z 
2026-05-03T08:02:49.7259216Z /opt/hostedtoolcache/Python/3.14.4/x64/lib/python3.14/site-packages/pytest_homeassistant_custom_component/plugins.py:436: Failed
2026-05-03T08:02:49.7259926Z ----------------------------- Captured stderr call -----------------------------
2026-05-03T08:02:49.7260419Z WARNING:custom_components.ramses_cc.services:Starting binding process for device 01:123456
2026-05-03T08:02:49.7261006Z WARNING:custom_components.ramses_cc.services:Success! Binding process completed for device 01:123456
2026-05-03T08:02:49.7261512Z ------------------------------ Captured log call -------------------------------
2026-05-03T08:02:49.7262297Z WARNING  custom_components.ramses_cc.services:services.py:81 Starting binding process for device 01:123456
2026-05-03T08:02:49.7262969Z WARNING  custom_components.ramses_cc.services:services.py:94 Success! Binding process completed for device 01:123456
2026-05-03T08:02:49.7263522Z --------------------------- snapshot report summary ----------------------------
2026-05-03T08:02:49.7263842Z 3 snapshots passed.
2026-05-03T08:02:49.7264086Z =========================== short test summary info ============================
2026-05-03T08:02:49.7265282Z ERROR tests/tests_new/test_services.py::test_bind_device_success - Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad74ac30>>>
2026-05-03T08:02:49.7267018Z ERROR tests/tests_new/test_services.py::test_send_packet_hgi_alias - Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad732150>>>
2026-05-03T08:02:49.7268804Z ERROR tests/tests_new/test_services.py::test_async_bind_device_routes_to_registry - Failed: Lingering timer after job <Job call_later 5 HassJobType.Executor <bound method RamsesServiceHandler._schedule_refresh of <custom_components.ramses_cc.services.RamsesServiceHandler object at 0x7f1cad7b28d0>>>
2026-05-03T08:02:49.7270149Z ============= 501 passed, 10 skipped, 3 errors in 65.25s (0:01:05) =============
2026-05-03T08:02:49.8419041Z ##[error]Process completed with exit code 1.

```

--- a/custom_components/ramses_cc/services.py
+++ b/custom_components/ramses_cc/services.py
@@ -7,7 +7,7 @@ import logging
 import re
 from typing import TYPE_CHECKING, Any, Final
 
-from homeassistant.core import ServiceCall
+from homeassistant.core import ServiceCall, callback
 from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
 from homeassistant.helpers import device_registry as dr, entity_registry as er
 from homeassistant.helpers.event import async_call_later
@@ -42,15 +42,13 @@ class RamsesServiceHandler:
         self._coordinator = coordinator
         self.hass = coordinator.hass
 
+    @callback
     def _schedule_refresh(self, _: Any) -> None:
         """Schedule a coordinator refresh.
 
         :param _: Unused argument (required for async_call_later callback signature).
         """
-        asyncio.run_coroutine_threadsafe(
-            self._coordinator.async_request_refresh(),
-            self.hass.loop,
-        )
+        self.hass.async_create_task(self._coordinator.async_request_refresh())
 
     async def async_bind_device(self, call: ServiceCall) -> None:
         """Handle the bind_device service call to bind a device to the system.

--- a/tests/tests_new/test_services.py
+++ b/tests/tests_new/test_services.py
@@ -326,7 +326,7 @@ async def test_bind_device_success(mock_coordinator: RamsesCoordinator) -> None:
     }
 
     # Should not raise exception
-    with patch("homeassistant.helpers.event.async_call_later"):
+    with patch.object(mock_coordinator.hass.loop, "call_later"):
         await mock_coordinator.async_bind_device(call)
 
     # Verify call later was scheduled
@@ -349,7 +349,7 @@ async def test_send_packet_hgi_alias(mock_coordinator: RamsesCoordinator) -> Non
         "payload": "FF",
     }
 
-    with patch("homeassistant.helpers.event.async_call_later"):
+    with patch.object(mock_coordinator.hass.loop, "call_later"):
         await mock_coordinator.async_send_packet(call)
 
     # Check that create_cmd was called with the REAL HGI ID, not the alias
@@ -2431,7 +2431,7 @@ async def test_async_bind_device_routes_to_registry(
     )
 
     # 2. Act: Execute the service
-    with patch("homeassistant.helpers.event.async_call_later"):
+    with patch.object(hass.loop, "call_later"):
         await handler.async_bind_device(call)
 
     # 3. Assert: Verify the registry was called, bypassing the Gateway

--- a/tests/tests_new/test_services.py
+++ b/tests/tests_new/test_services.py
@@ -14,6 +14,7 @@ from homeassistant.helpers import device_registry as dr, entity_registry as er
 from homeassistant.util import dt as dt_util
 from pytest_homeassistant_custom_component.common import (  # type: ignore[import-untyped]
     MockConfigEntry,
+    async_fire_time_changed,
 )
 
 from custom_components.ramses_cc.const import (
@@ -326,11 +327,11 @@ async def test_bind_device_success(mock_coordinator: RamsesCoordinator) -> None:
     }
 
     # Should not raise exception
-    with patch("custom_components.ramses_cc.services.async_call_later"):
-        await mock_coordinator.async_bind_device(call)
+    await mock_coordinator.async_bind_device(call)
 
-        # Timer is mocked, wait for event loop to finish cleanly
-        await mock_coordinator.hass.async_block_till_done()
+    # Fast-forward time to cleanly execute and remove the async_call_later timer
+    async_fire_time_changed(mock_coordinator.hass, dt_util.utcnow() + td(seconds=10))
+    await mock_coordinator.hass.async_block_till_done()
 
     # Verify call later was scheduled
     assert mock_device._initiate_binding_process.called
@@ -352,11 +353,11 @@ async def test_send_packet_hgi_alias(mock_coordinator: RamsesCoordinator) -> Non
         "payload": "FF",
     }
 
-    with patch("custom_components.ramses_cc.services.async_call_later"):
-        await mock_coordinator.async_send_packet(call)
+    await mock_coordinator.async_send_packet(call)
 
-        # Timer is mocked, wait for event loop to finish cleanly
-        await mock_coordinator.hass.async_block_till_done()
+    # Fast-forward time to cleanly execute and remove the async_call_later timer
+    async_fire_time_changed(mock_coordinator.hass, dt_util.utcnow() + td(seconds=10))
+    await mock_coordinator.hass.async_block_till_done()
 
     # Check that create_cmd was called with the REAL HGI ID, not the alias
     # This verifies the translation logic
@@ -1870,8 +1871,8 @@ async def test_get_fan_param_raises_error_missing_destination(
         await mock_coordinator.async_get_fan_param(call_data)
 
 
-async def test_schedule_refresh_threadsafe(mock_coordinator: MagicMock) -> None:
-    """Test that _schedule_refresh submits the refresh request to the loop thread-safely."""
+async def test_schedule_refresh_creates_task(mock_coordinator: MagicMock) -> None:
+    """Test that _schedule_refresh submits the refresh request as a task."""
 
     # 1. Mock the coordinator's refresh method so we can assert it was called
     # We use AsyncMock so it returns a coroutine object when called, just like the real method
@@ -1880,25 +1881,20 @@ async def test_schedule_refresh_threadsafe(mock_coordinator: MagicMock) -> None:
     # 2. Instantiate the handler with the mock coordinator
     handler = RamsesServiceHandler(mock_coordinator)
 
-    # 3. Patch run_coroutine_threadsafe to intercept the call
-    with patch(
-        "custom_components.ramses_cc.services.asyncio.run_coroutine_threadsafe"
-    ) as mock_run_threadsafe:
+    # 3. Patch async_create_task to intercept the call
+    with patch.object(mock_coordinator.hass, "async_create_task") as mock_create_task:
         # 4. Trigger the method (it expects one argument, usually a datetime)
         handler._schedule_refresh(None)
 
         # 5. Verify the coordinator's refresh method was called to generate the coroutine
         mock_coordinator.async_request_refresh.assert_called_once()
 
-        # 6. Verify the coroutine was submitted to the threadsafe runner
-        mock_run_threadsafe.assert_called_once()
+        # 6. Verify the coroutine was submitted to the task runner
+        mock_create_task.assert_called_once()
 
-        # Check arguments: (coroutine_object, event_loop)
-        args, _ = mock_run_threadsafe.call_args
+        # Check arguments
+        args, _ = mock_create_task.call_args
         coro_arg = args[0]
-        loop_arg = args[1]
-
-        assert loop_arg == mock_coordinator.hass.loop
 
         # Cleanup: Prevent "RuntimeWarning: coroutine '...' was never awaited"
         # Since we intercepted it, it won't run, so we close it manually.
@@ -2437,11 +2433,11 @@ async def test_async_bind_device_routes_to_registry(
     )
 
     # 2. Act: Execute the service
-    with patch("custom_components.ramses_cc.services.async_call_later"):
-        await handler.async_bind_device(call)
+    await handler.async_bind_device(call)
 
-        # Timer is mocked, wait for event loop to finish cleanly
-        await hass.async_block_till_done()
+    # Fast-forward time to cleanly execute and remove the async_call_later timer
+    async_fire_time_changed(hass, dt_util.utcnow() + td(seconds=10))
+    await hass.async_block_till_done()
 
     # 3. Assert: Verify the registry was called, bypassing the Gateway
     mock_registry.fake_device.assert_called_once_with("01:123456")

--- a/tests/tests_new/test_services.py
+++ b/tests/tests_new/test_services.py
@@ -326,7 +326,7 @@ async def test_bind_device_success(mock_coordinator: RamsesCoordinator) -> None:
     }
 
     # Should not raise exception
-    with patch("homeassistant.helpers.event.async_call_later"):
+    with patch("custom_components.ramses_cc.services.async_call_later"):
         await mock_coordinator.async_bind_device(call)
 
         # Timer is mocked, wait for event loop to finish cleanly
@@ -352,7 +352,7 @@ async def test_send_packet_hgi_alias(mock_coordinator: RamsesCoordinator) -> Non
         "payload": "FF",
     }
 
-    with patch("homeassistant.helpers.event.async_call_later"):
+    with patch("custom_components.ramses_cc.services.async_call_later"):
         await mock_coordinator.async_send_packet(call)
 
         # Timer is mocked, wait for event loop to finish cleanly
@@ -2437,7 +2437,7 @@ async def test_async_bind_device_routes_to_registry(
     )
 
     # 2. Act: Execute the service
-    with patch("homeassistant.helpers.event.async_call_later"):
+    with patch("custom_components.ramses_cc.services.async_call_later"):
         await handler.async_bind_device(call)
 
         # Timer is mocked, wait for event loop to finish cleanly

--- a/tests/tests_new/test_services.py
+++ b/tests/tests_new/test_services.py
@@ -14,6 +14,7 @@ from homeassistant.helpers import device_registry as dr, entity_registry as er
 from homeassistant.util import dt as dt_util
 from pytest_homeassistant_custom_component.common import (  # type: ignore[import-untyped]
     MockConfigEntry,
+    async_fire_time_changed,
 )
 
 from custom_components.ramses_cc.const import (
@@ -326,8 +327,14 @@ async def test_bind_device_success(mock_coordinator: RamsesCoordinator) -> None:
     }
 
     # Should not raise exception
-    with patch.object(mock_coordinator.hass.loop, "call_later"):
+    with patch.object(mock_coordinator.service_handler, "_schedule_refresh"):
         await mock_coordinator.async_bind_device(call)
+
+        # Fast-forward time to cleanly execute and remove the async_call_later timer
+        async_fire_time_changed(
+            mock_coordinator.hass, dt_util.utcnow() + td(seconds=10)
+        )
+        await mock_coordinator.hass.async_block_till_done()
 
     # Verify call later was scheduled
     assert mock_device._initiate_binding_process.called
@@ -349,8 +356,14 @@ async def test_send_packet_hgi_alias(mock_coordinator: RamsesCoordinator) -> Non
         "payload": "FF",
     }
 
-    with patch.object(mock_coordinator.hass.loop, "call_later"):
+    with patch.object(mock_coordinator.service_handler, "_schedule_refresh"):
         await mock_coordinator.async_send_packet(call)
+
+        # Fast-forward time to cleanly execute and remove the async_call_later timer
+        async_fire_time_changed(
+            mock_coordinator.hass, dt_util.utcnow() + td(seconds=10)
+        )
+        await mock_coordinator.hass.async_block_till_done()
 
     # Check that create_cmd was called with the REAL HGI ID, not the alias
     # This verifies the translation logic
@@ -2431,8 +2444,12 @@ async def test_async_bind_device_routes_to_registry(
     )
 
     # 2. Act: Execute the service
-    with patch.object(hass.loop, "call_later"):
+    with patch.object(handler, "_schedule_refresh"):
         await handler.async_bind_device(call)
+
+        # Fast-forward time to cleanly execute and remove the async_call_later timer
+        async_fire_time_changed(hass, dt_util.utcnow() + td(seconds=10))
+        await hass.async_block_till_done()
 
     # 3. Assert: Verify the registry was called, bypassing the Gateway
     mock_registry.fake_device.assert_called_once_with("01:123456")

--- a/tests/tests_new/test_services.py
+++ b/tests/tests_new/test_services.py
@@ -326,7 +326,7 @@ async def test_bind_device_success(mock_coordinator: RamsesCoordinator) -> None:
     }
 
     # Should not raise exception
-    with patch("custom_components.ramses_cc.services.async_call_later"):
+    with patch("homeassistant.helpers.event.async_call_later"):
         await mock_coordinator.async_bind_device(call)
 
     # Verify call later was scheduled
@@ -349,7 +349,7 @@ async def test_send_packet_hgi_alias(mock_coordinator: RamsesCoordinator) -> Non
         "payload": "FF",
     }
 
-    with patch("custom_components.ramses_cc.services.async_call_later"):
+    with patch("homeassistant.helpers.event.async_call_later"):
         await mock_coordinator.async_send_packet(call)
 
     # Check that create_cmd was called with the REAL HGI ID, not the alias
@@ -2431,7 +2431,7 @@ async def test_async_bind_device_routes_to_registry(
     )
 
     # 2. Act: Execute the service
-    with patch("custom_components.ramses_cc.services.async_call_later"):
+    with patch("homeassistant.helpers.event.async_call_later"):
         await handler.async_bind_device(call)
 
     # 3. Assert: Verify the registry was called, bypassing the Gateway

--- a/tests/tests_new/test_services.py
+++ b/tests/tests_new/test_services.py
@@ -326,7 +326,8 @@ async def test_bind_device_success(mock_coordinator: RamsesCoordinator) -> None:
     }
 
     # Should not raise exception
-    await mock_coordinator.async_bind_device(call)
+    with patch("custom_components.ramses_cc.services.async_call_later"):
+        await mock_coordinator.async_bind_device(call)
 
     # Verify call later was scheduled
     assert mock_device._initiate_binding_process.called
@@ -348,7 +349,8 @@ async def test_send_packet_hgi_alias(mock_coordinator: RamsesCoordinator) -> Non
         "payload": "FF",
     }
 
-    await mock_coordinator.async_send_packet(call)
+    with patch("custom_components.ramses_cc.services.async_call_later"):
+        await mock_coordinator.async_send_packet(call)
 
     # Check that create_cmd was called with the REAL HGI ID, not the alias
     # This verifies the translation logic
@@ -2429,7 +2431,8 @@ async def test_async_bind_device_routes_to_registry(
     )
 
     # 2. Act: Execute the service
-    await handler.async_bind_device(call)
+    with patch("custom_components.ramses_cc.services.async_call_later"):
+        await handler.async_bind_device(call)
 
     # 3. Assert: Verify the registry was called, bypassing the Gateway
     mock_registry.fake_device.assert_called_once_with("01:123456")

--- a/tests/tests_new/test_services.py
+++ b/tests/tests_new/test_services.py
@@ -326,12 +326,15 @@ async def test_bind_device_success(mock_coordinator: RamsesCoordinator) -> None:
         "device_info": None,
     }
 
-    # Should not raise exception
-    await mock_coordinator.async_bind_device(call)
+    # Intercept the refresh request so Debouncers are never spawned
+    with patch.object(mock_coordinator, "async_request_refresh"):
+        await mock_coordinator.async_bind_device(call)
 
-    # Fast-forward time to cleanly execute and remove the async_call_later timer
-    async_fire_time_changed(mock_coordinator.hass, dt_util.utcnow() + td(seconds=10))
-    await mock_coordinator.hass.async_block_till_done()
+        # Fast-forward time to cleanly execute the async_call_later timer
+        async_fire_time_changed(
+            mock_coordinator.hass, dt_util.utcnow() + td(seconds=10)
+        )
+        await mock_coordinator.hass.async_block_till_done()
 
     # Verify call later was scheduled
     assert mock_device._initiate_binding_process.called
@@ -353,11 +356,15 @@ async def test_send_packet_hgi_alias(mock_coordinator: RamsesCoordinator) -> Non
         "payload": "FF",
     }
 
-    await mock_coordinator.async_send_packet(call)
+    # Intercept the refresh request so Debouncers are never spawned
+    with patch.object(mock_coordinator, "async_request_refresh"):
+        await mock_coordinator.async_send_packet(call)
 
-    # Fast-forward time to cleanly execute and remove the async_call_later timer
-    async_fire_time_changed(mock_coordinator.hass, dt_util.utcnow() + td(seconds=10))
-    await mock_coordinator.hass.async_block_till_done()
+        # Fast-forward time to cleanly execute the async_call_later timer
+        async_fire_time_changed(
+            mock_coordinator.hass, dt_util.utcnow() + td(seconds=10)
+        )
+        await mock_coordinator.hass.async_block_till_done()
 
     # Check that create_cmd was called with the REAL HGI ID, not the alias
     # This verifies the translation logic
@@ -2397,6 +2404,8 @@ async def test_async_bind_device_routes_to_registry(
     # 1. Arrange: Setup Coordinator
     mock_coordinator = MagicMock()
     mock_coordinator.hass = hass
+    # Prevent HA Debouncer by mocking the refresh request entirely
+    mock_coordinator.async_request_refresh = AsyncMock()
 
     # Setup the client (Gateway) mock
     mock_client = MagicMock()
@@ -2435,7 +2444,7 @@ async def test_async_bind_device_routes_to_registry(
     # 2. Act: Execute the service
     await handler.async_bind_device(call)
 
-    # Fast-forward time to cleanly execute and remove the async_call_later timer
+    # Fast-forward time to cleanly execute the async_call_later timer
     async_fire_time_changed(hass, dt_util.utcnow() + td(seconds=10))
     await hass.async_block_till_done()
 

--- a/tests/tests_new/test_services.py
+++ b/tests/tests_new/test_services.py
@@ -14,7 +14,6 @@ from homeassistant.helpers import device_registry as dr, entity_registry as er
 from homeassistant.util import dt as dt_util
 from pytest_homeassistant_custom_component.common import (  # type: ignore[import-untyped]
     MockConfigEntry,
-    async_fire_time_changed,
 )
 
 from custom_components.ramses_cc.const import (
@@ -327,13 +326,10 @@ async def test_bind_device_success(mock_coordinator: RamsesCoordinator) -> None:
     }
 
     # Should not raise exception
-    with patch.object(mock_coordinator.service_handler, "_schedule_refresh"):
+    with patch("homeassistant.helpers.event.async_call_later"):
         await mock_coordinator.async_bind_device(call)
 
-        # Fast-forward time to cleanly execute and remove the async_call_later timer
-        async_fire_time_changed(
-            mock_coordinator.hass, dt_util.utcnow() + td(seconds=10)
-        )
+        # Timer is mocked, wait for event loop to finish cleanly
         await mock_coordinator.hass.async_block_till_done()
 
     # Verify call later was scheduled
@@ -356,13 +352,10 @@ async def test_send_packet_hgi_alias(mock_coordinator: RamsesCoordinator) -> Non
         "payload": "FF",
     }
 
-    with patch.object(mock_coordinator.service_handler, "_schedule_refresh"):
+    with patch("homeassistant.helpers.event.async_call_later"):
         await mock_coordinator.async_send_packet(call)
 
-        # Fast-forward time to cleanly execute and remove the async_call_later timer
-        async_fire_time_changed(
-            mock_coordinator.hass, dt_util.utcnow() + td(seconds=10)
-        )
+        # Timer is mocked, wait for event loop to finish cleanly
         await mock_coordinator.hass.async_block_till_done()
 
     # Check that create_cmd was called with the REAL HGI ID, not the alias
@@ -2444,11 +2437,10 @@ async def test_async_bind_device_routes_to_registry(
     )
 
     # 2. Act: Execute the service
-    with patch.object(handler, "_schedule_refresh"):
+    with patch("homeassistant.helpers.event.async_call_later"):
         await handler.async_bind_device(call)
 
-        # Fast-forward time to cleanly execute and remove the async_call_later timer
-        async_fire_time_changed(hass, dt_util.utcnow() + td(seconds=10))
+        # Timer is mocked, wait for event loop to finish cleanly
         await hass.async_block_till_done()
 
     # 3. Assert: Verify the registry was called, bypassing the Gateway

--- a/tests/tests_old/test_evo_control.py
+++ b/tests/tests_old/test_evo_control.py
@@ -18,7 +18,7 @@ This does not test any service calls, or any other endpoints.
 from __future__ import annotations
 
 import json
-from typing import Any
+from typing import Any, cast
 
 from homeassistant.components.binary_sensor import BinarySensorEntity
 from homeassistant.components.climate import ClimateEntity
@@ -79,7 +79,7 @@ async def instantiate_entities(
     # have to stop MessageIndex thread, aka: gwy.msg_db.stop()
     await gwy.stop()
 
-    coordinator: RamsesCoordinator = MockRamsesCoordinator(hass)
+    coordinator = cast(RamsesCoordinator, MockRamsesCoordinator(hass))
 
     # climate entities (TCS, zones) - UPDATED to use device_registry
     rf_climates = [s for s in gwy.device_registry.systems if isinstance(s, Evohome)]
@@ -91,7 +91,11 @@ async def instantiate_entities(
     ]
 
     climates: list[ClimateEntity] = [
-        description.ramses_cc_class(coordinator, device, description)
+        description.ramses_cc_class(
+            coordinator,
+            cast(Any, device),
+            description,
+        )
         for device in rf_climates
         for description in CLIMATE_DESCRIPTIONS
         if isinstance(device, description.ramses_rf_class)
@@ -103,7 +107,11 @@ async def instantiate_entities(
     ]
 
     water_heaters: list[WaterHeaterEntity] = [
-        description.ramses_cc_class(coordinator, device, description)
+        description.ramses_cc_class(
+            coordinator,
+            cast(Any, device),
+            description,
+        )
         for device in rf_heaters
         for description in WATER_HEATER_DESCRIPTIONS
         if isinstance(device, description.ramses_rf_class)
@@ -113,7 +121,11 @@ async def instantiate_entities(
     rf_devices = list(gwy.device_registry.devices) + rf_climates + rf_heaters
 
     binary_sensors: list[BinarySensorEntity] = [
-        description.ramses_cc_class(coordinator, device, description)
+        description.ramses_cc_class(
+            coordinator,
+            cast(Any, device),
+            description,
+        )
         for device in rf_devices
         for description in BINARY_SENSOR_DESCRIPTIONS
         if isinstance(device, description.ramses_rf_class)
@@ -121,7 +133,11 @@ async def instantiate_entities(
     ]
 
     sensors: list[SensorEntity] = [
-        description.ramses_cc_class(coordinator, device, description)
+        description.ramses_cc_class(
+            coordinator,
+            cast(Any, device),
+            description,
+        )
         for device in rf_devices
         for description in SENSOR_DESCRIPTIONS
         if isinstance(device, description.ramses_rf_class)
@@ -167,7 +183,9 @@ async def test_namespace(hass: HomeAssistant) -> None:
 
     #
     # evo_control uses: the working_schema
-    schema = binary.extra_state_attributes["working_schema"]
+    attrs = binary.extra_state_attributes
+    assert attrs is not None
+    schema = attrs["working_schema"]
     assert schema["stored_hotwater"] == SCHEMA["stored_hotwater"]
     assert schema["zones"] == SCHEMA["zones"]
 
@@ -185,7 +203,9 @@ async def test_namespace(hass: HomeAssistant) -> None:
         assert binary.unique_id == uid
         assert binary.state in (STATE_ON, STATE_OFF, None)
 
-        battery_level = binary.extra_state_attributes["battery_level"]
+        attrs = binary.extra_state_attributes
+        assert attrs is not None
+        battery_level = attrs["battery_level"]
         assert (
             battery_level is None or 0.0 < battery_level < 1.0
         )  # TODO: check values, not types
@@ -242,11 +262,12 @@ async def test_namespace(hass: HomeAssistant) -> None:
 
     assert climate.state == HVACMode.HEAT
     assert climate.preset_mode == PRESET_ECO
-    assert climate.extra_state_attributes["system_mode"]["system_mode"] == "eco_boost"
-    assert (
-        as_iso(climate.extra_state_attributes["system_mode"]["until"])
-        == "2022-03-06T14:44:00"
-    )
+
+    attrs = climate.extra_state_attributes
+    assert attrs is not None
+    sys_mode = attrs["system_mode"]
+    assert sys_mode["system_mode"] == "eco_boost"
+    assert as_iso(sys_mode["until"]) == "2022-03-06T14:44:00"
 
     #
     # evo_control uses: climate.${cid}_${haZid}
@@ -258,8 +279,11 @@ async def test_namespace(hass: HomeAssistant) -> None:
         # assert climate.name == SCHEMA["zones"][zon_idx]["_name"]
         # TODO
 
+        attrs = climate.extra_state_attributes
+        assert attrs is not None
+
         if zon_idx == "02":
-            assert climate.extra_state_attributes["mode"] == {
+            assert attrs["mode"] == {
                 "mode": "permanent_override",
                 "setpoint": 5.0,
             }
@@ -268,7 +292,7 @@ async def test_namespace(hass: HomeAssistant) -> None:
             assert climate.current_temperature == 18.16
 
         else:
-            mode_attr = climate.extra_state_attributes["mode"]
+            mode_attr = attrs["mode"]
             assert mode_attr["mode"] == "temporary_override"
             assert mode_attr["setpoint"] == 20.0
             # Convert datetime to string for the assertion
@@ -283,7 +307,9 @@ async def test_namespace(hass: HomeAssistant) -> None:
     assert heater.unique_id == uid
     # assert heater.name == f"{CTL_ID} XXX"  # TODO set name
 
-    heater_mode = heater.extra_state_attributes["mode"]
+    attrs = heater.extra_state_attributes
+    assert attrs is not None
+    heater_mode = attrs["mode"]
     assert heater_mode["mode"] == "temporary_override"
     assert heater_mode["active"] is True
     assert as_iso(heater_mode["until"]) == "2022-02-10T22:00:00"

